src/site/content/en/blog/hands-on-portals/index.md

```diff
@@ -0,0 +1,196 @@
+
+![A screenshot of the DevTools console showing the HTMLPortalElement](HTMLPortalElement.png)
+
+Let’s walk through a basic example.
+
+```javascript
+// Create a portal with the wikipedia page, and embed it
+// (like an iframe). You can also use <portal> tag instead.
+portal = document.createElement('portal');
+portal.src = 'https://en.wikipedia.org/wiki/World_Wide_Web';
+portal.style = '...';
+document.body.appendChild(portal);
+
+// When the user touches the preview (embedded portal):
+// do fancy animation, e.g. expand …
+// and finish by doing the actual transition
+portal.activate();
+```
+
+It’s that simple. Try this code in the DevTools console, the wikipedia page should open up. If you wanted to build something like we showed in CDS, the following snippet will be of interest.
+
+![A gif of preview portal style demo](portal-preview-demo.gif)
+
+```javascript
+// Adding some styles with transitions
+const style = document.createElement('style');
+const initialScale = 0.4;
+style.innerHTML = `
+  portal {
+    position:fixed;
+    width: 100%;
+    height: 100%;
+    opacity: 0;
+    transition:
+      transform 0.4s,
+      bottom 0.7s,
+      left 0.7s,
+      opacity 1.0s;
+    box-shadow: 0 0 20px 10px #999;
+    transform: scale(${initialScale});
+    bottom: calc(20px + 50% * ${initialScale} - 50%);
+    left: calc(20px + 50% * ${initialScale} - 50%);
+    z-index: 10000;
+  }
+  .portal-reveal {
+    transform: scale(1.0);
+    bottom: 0px;
+    left: 0px;
+  }
+  .fade-in {
+    opacity: 1.0;
+  }
+`;
+const portal = document.createElement('portal');
+// Let’s navigate into the WICG Portals spec page
+portal.src = 'https://wicg.github.io/portals/';
+portal.addEventListener('click', evt => {
+  // Animate the portal once user interacts
+  portal.classList.add('portal-reveal');
+});
+portal.addEventListener('transitionend', evt => {
+  if (evt.propertyName == 'bottom') {
+    // Activate the portal once the transition has completed
+    portal.activate();
+  }
+});
+document.body.append(style, portal);
+
+// Waiting for the page to load.
+// using setTimeout is a suboptimal way and it’s best to fade-in
+// when receiving a load complete message from the portal via postMessage
+setTimeout(_ => portal.classList.add('fade-in'), 2000);
+```
+
+It is also easy to do feature detection to progressively enhance a website using Portals.
+
+```javascript
+if ('HTMLPortalElement' in window) {
+  // If this is a platform that has Portals...
+  const portal = document.createElement('portal');
+  ...
+}
+```
+
+If you want to quickly experience what Portals feels like, try using [uskay-portals-demo.glitch.me](https://uskay-portals-demo.glitch.me). 1) Enter a URL you want to preview, 2) the page will then be embedded as a portal element, 3) click on the preview 4) The preview will be activated after an animation.
+
+![A gif of using the glitch demo](glitch.gif)
```

Comment:
Suggest wrapping this with the `w-screenshot` class.

Reply:
I will do that!